python/cudf/cudf/core/frame.py

```diff
@@ -1031,23 +1083,104 @@ def dropna(self, axis=0, how="any", subset=None, thresh=None):
-            less than `thresh` non-null values
-
+        inplace : bool, default False
+            If True, do operation inplace and return None.
 
         Returns
         -------
         Copy of the DataFrame with rows/columns containing nulls dropped.
+
+        See also
+        --------
+        cudf.core.dataframe.DataFrame.isna
+            Indicate null values.
+
+        cudf.core.dataframe.DataFrame.notna
+            Indicate non-null values.
+
+        cudf.core.dataframe.DataFrame.fillna
+            Replace null values.
+
+        cudf.core.series.Series.dropna
+            Drop null values.
+
+        cudf.core.index.Index.dropna
+            Drop null indices.
+
+        Examples
+        --------
+        >>> import cudf
+        >>> df = cudf.DataFrame({"name": ['Alfred', 'Batman', 'Catwoman'],
+        ...                    "toy": ['Batmobile', None, 'Bullwhip'],
+        ...                    "born": [np.datetime64("1940-04-25"),
+        ...                             np.datetime64("NaT"),
+        ...                             np.datetime64("NaT")]})
+        >>> df
+               name        toy       born
+        0    Alfred  Batmobile 1940-04-25
+        1    Batman       None       null
+        2  Catwoman   Bullwhip       null
+
+        Drop the rows where at least one element is null.
+
+        >>> df.dropna()
+             name        toy       born
+        0  Alfred  Batmobile 1940-04-25
+
+        Drop the columns where at least one element is null.
+
+        >>> df.dropna(axis='columns')
+               name
+        0    Alfred
+        1    Batman
+        2  Catwoman
+
+        Drop the rows where all elements are null.
+
+        >>> df.dropna(how='all')
+               name        toy       born
+        0    Alfred  Batmobile 1940-04-25
+        1    Batman       None       null
+        2  Catwoman   Bullwhip       null
+
+        Keep only the rows with at least 2 non-null values.
+
+        >>> df.dropna(thresh=2)
+               name        toy       born
+        0    Alfred  Batmobile 1940-04-25
+        2  Catwoman   Bullwhip       null
+
+        Define in which columns to look for null values.
+
+        >>> df.dropna(subset=['name', 'born'])
+             name        toy       born
+        0  Alfred  Batmobile 1940-04-25
+
+        Keep the DataFrame with valid entries in the same variable.
+
+        >>> df.dropna(inplace=True)
+        >>> df
+             name        toy       born
+        0  Alfred  Batmobile 1940-04-25
         """
         if axis == 0:
-            return self._drop_na_rows(how=how, subset=subset, thresh=thresh)
+            result = self._drop_na_rows(how=how, subset=subset, thresh=thresh)
         else:
-            return self._drop_na_columns(how=how, subset=subset, thresh=thresh)
+            result = self._drop_na_columns(
+                how=how, subset=subset, thresh=thresh
+            )
+
+        if isinstance(self, cudf.DataFrame):
```

Comment:
If we do need to specialize for a subclass like this, we should override `dropna` in the subclass and call `super()`. If we do that, does the docstring for `Frame.dropna` appear under `DataFrame` in the sphinx generated docs? It certainly does in IPython for instance:

```python

In [18]: class Foo():
    ...:     def method(self):
    ...:         """
    ...:         My docstring
    ...:         """
    ...:         pass
    ...:

In [19]: class Bar(Foo):
    ...:     def method(self):
    ...:         super().method()
    ...:

In [20]: help(Bar.method)

Help on function method in module __main__:

method(self)
    My docstring
```

Reply:
Actually this specialization is not required, cleaned it up. This probably made its way through because of some intermediate changes I was making.



> If we do that, does the docstring for `Frame.dropna` appear under `DataFrame` in the sphinx generated docs? 

Yes it surfaces in sphinx docs too if we choose to do that.